New UD Annotatrix logo

Persevere04 opened the pull request:
![Ggtee](https://user-images.githubusercontent.com/58814736/70850211-acfa8300-1e7f-11ea-8fc9-613319d0aaad.png)

Commit:
only show the info icon if we are lossy on at least one field

Commit:
latest version of notatrix & rebuild

ftyers committed:
orthography

Commit:
fix empty scroll bar issue on firefox linux

Commit:
fix paths in "modes"

Commit:
fixing latex export

Commit:
better model for Rooms that (hopefully) won't crash as frequently

Commit:
fix progress bar issue and invalid edge issue

Commit:
reimplement the cycle detector

Commit:
only draw one dependency if we're not enhanced

Commit:
make sure to re-enable tabs once we make their format valid again

Commit:
some tests for dependency validation (not complete)

Commit:
get rid of awkward scroll bar on the #status container in linux chromium

Commit:
latest notatrix version and rebuild

keggsmurph21 committed:
First draft of the new README, some things are just outlined

keggsmurph21 committed:
formatting fixes

keggsmurph21 committed:
some how-to-use fields filled in

jonorthwash committed:
Update README.md

fix typo

diogoscf committed:
Fix table view deleting words

keggsmurph21 committed:
Merge pull request #327 from diogoscf/master

Fix table view deleting words

diogoscf committed:
Fix pressing enter disabling navigation

jonorthwash committed:
Merge pull request #329 from diogoscf/master

Fix pressing enter disabling navigation

diogoscf committed:
Fixes table view not showing dependencies

diogoscf committed:
Fix text disappearing when cloumn is expanded

jonorthwash committed:
Merge pull request #332 from diogoscf/master

Fix table view not showing dependencies

diogoscf committed:
Fix Autocomplete box width

jonorthwash committed:
Merge pull request #336 from diogoscf/master

Fix Autocomplete box width

keggsmurph21 committed:
small changes to the scraper for better output

keggsmurph21 committed:
remove old dependencies

keggsmurph21 committed:
move the ud scraper out of this project and into another (keggsmurph21/notatrix-utils)

keggsmurph21 committed:
add partially finished install script, needs work still

Commit:
uncommitted changes

Commit:
do merge

Commit:
add more Client documentation, build that documentation into Markdown file, add gulp task to build the docs again

yaskevich committed:
update dependencies to actual ones

It allows to successfully complete `npm install` run all the build in up-to-date environment.

yaskevich committed:
chat state change

now chat window is minimized by default

jonorthwash committed:
Merge pull request #342 from yaskevich/master

Fix outdated packages deps

yaskevich committed:
rebuilt bundle.js

yaskevich committed:
migration to gulp 4

yaskevich committed:
babel preset fix

jonorthwash committed:
Merge pull request #348 from yaskevich/master

rebuilt bundle.js

yaskevich committed:
fixing work in static/offline mode

yaskevich committed:
modals on Windows (ejs bug)

yaskevich committed:
upload in standalone mode, UI fixes, build minification

yaskevich committed:
UI, build fixes, error handling

yaskevich committed:
code reindent and minor fixes in UI

jonorthwash committed:
updated readme with beginnings of feature grid

jonorthwash committed:
oops, table was wrong

jonorthwash committed:
examples of most uses

jonorthwash committed:
better table?

jonorthwash committed:
betterer table?

yaskevich committed:
small fix: differentiating upload from main and internal page

yaskevich committed:
tabs in index.ejs

yaskevich committed:
tabs in index.html

jonorthwash committed:
Merge pull request #356 from yaskevich/master

Static mode, server, building, UI/UX

dependabot[bot] committed:
Bump mixin-deep from 1.3.1 to 1.3.2

Bumps [mixin-deep](https://github.com/jonschlinkert/mixin-deep) from 1.3.1 to 1.3.2.
- [Release notes](https://github.com/jonschlinkert/mixin-deep/releases)
- [Commits](https://github.com/jonschlinkert/mixin-deep/compare/1.3.1...1.3.2)

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Fix #381: prevent corpus modifications from resetting the index

If two users were editing different sentences on the same treebank
in different windows connected to the same backend, then making a
change in one window would cause the `app.corpus.index` on the other
window to be reset (set to 0).  This meant that whenever one user
made a change, the other user would have to navigate all the way
back to their place.

This bug was happening because whenever we re-loaded the corpus via
the `/save` ajax path, we would broadcast an "index" of 0 to all the
other connected users, since the `/save` path is meant for entire-
corpus overwrites.  The app would then interpret this index as the
current index for the window, and then reset it.

This patch introduces a simple fix: save the current index before
we overwrite the corpus, and then just navigate to it after overwriting.

Also included are rebuilt bundles and an updated package-lock.json.

dependabot[bot] committed:
Bump cached-path-relative from 1.0.1 to 1.0.2

Bumps [cached-path-relative](https://github.com/ashaffer/cached-path-relative) from 1.0.1 to 1.0.2.
- [Release notes](https://github.com/ashaffer/cached-path-relative/releases)
- [Commits](https://github.com/ashaffer/cached-path-relative/commits)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump extend from 3.0.1 to 3.0.2

Bumps [extend](https://github.com/justmoon/node-extend) from 3.0.1 to 3.0.2.
- [Release notes](https://github.com/justmoon/node-extend/releases)
- [Changelog](https://github.com/justmoon/node-extend/blob/master/CHANGELOG.md)
- [Commits](https://github.com/justmoon/node-extend/compare/v3.0.1...v3.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump lodash from 4.17.10 to 4.17.15

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.10 to 4.17.15.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.10...4.17.15)

Signed-off-by: dependabot[bot] <support@github.com>

jonorthwash committed:
Create README.md

kmurphy4 commented:
Hey @Persevere04 , thanks for the PR. It looks like there's something up with the diff though (173k additions, 49k deletions). Could you either rebase off master or merge master into your branch and then reopen the PR?

keggsmurph21 committed:
keyboard: Avoid setting cursor to textarea end on change

Fixes #393

keggsmurph21 committed:
Merge branch 'master' of https://github.com/jonorthwash/ud-annotatrix

dependabot[bot] committed:
Bump acorn from 5.7.1 to 5.7.4

Bumps [acorn](https://github.com/acornjs/acorn) from 5.7.1 to 5.7.4.
- [Release notes](https://github.com/acornjs/acorn/releases)
- [Commits](https://github.com/acornjs/acorn/compare/5.7.1...5.7.4)

Signed-off-by: dependabot[bot] <support@github.com>

jonpan4 committed:
Fix deprel offset when changing to rtl.

keggsmurph21 committed:
App: Never try to download a "null" format

If the sentence is not "parsed", then it's possible that we could
have `app.corpus.format` return `null`.  However, the `Sentence::to()`
method expects that its `format` argument will be the key of a mapping
it maintains to generator functions.  In the case of a `null` format,
we'll get an "undefined is not a function"-type error and die.

This patch makes it so an "unparsed" sentence will always just spit
out plain text.  (Ideally this should also warn the user or
something...)

kmurphy4 committed:
upload: Don't try to POST if not running with a server backend

kmurphy4 committed:
notatrix: Bump to v2.2.2 (fixes #397)

keggsmurph21 committed:
Chat: Don't display on startup by default

keggsmurph21 committed:
Keyboard: Add missing key names

keggsmurph21 committed:
GUI: Support setting a node as "empty" by pressing "N"

Closes #403

dependabot[bot] committed:
Bump jquery from 3.3.1 to 3.5.0

Bumps [jquery](https://github.com/jquery/jquery) from 3.3.1 to 3.5.0.
- [Release notes](https://github.com/jquery/jquery/releases)
- [Commits](https://github.com/jquery/jquery/compare/3.3.1...3.5.0)

Signed-off-by: dependabot[bot] <support@github.com>

jonorthwash committed:
Merge pull request #395 from JPJPJPOPOP/master

Fix deprel offset when changing to rtl.

jonorthwash committed:
Merge pull request #380 from jonorthwash/dependabot/npm_and_yarn/mixin-deep-1.3.2

Bump mixin-deep from 1.3.1 to 1.3.2

jonorthwash committed:
Merge pull request #383 from jonorthwash/dependabot/npm_and_yarn/cached-path-relative-1.0.2

Bump cached-path-relative from 1.0.1 to 1.0.2

jonorthwash committed:
Merge pull request #384 from jonorthwash/dependabot/npm_and_yarn/extend-3.0.2

Bump extend from 3.0.1 to 3.0.2

jonorthwash committed:
Merge pull request #385 from jonorthwash/dependabot/npm_and_yarn/lodash-4.17.15

Bump lodash from 4.17.10 to 4.17.15

jonorthwash committed:
Merge pull request #394 from jonorthwash/dependabot/npm_and_yarn/acorn-5.7.4

Bump acorn from 5.7.1 to 5.7.4

jonorthwash committed:
Merge pull request #399 from jonorthwash/dependabot/npm_and_yarn/jquery-3.5.0

Bump jquery from 3.3.1 to 3.5.0

JPJPJPOPOP committed:
Inserting svg.

JPJPJPOPOP committed:
Add tokens to svg.

JPJPJPOPOP committed:
Removed cytoscape graph.

JPJPJPOPOP committed:
Trying to get adding deprels to work and changing branch.

JPJPJPOPOP committed:
Adding deprels works, but doesn't show in graph yet.

JPJPJPOPOP committed:
Deprels now show on graph. Need to add shifting of nodes.

JPJPJPOPOP committed:
Fix consistency of sizes after zooming and add rtl.

JPJPJPOPOP committed:
Edit box shows now and added shifting of nodes to account for the deprel label.

JPJPJPOPOP committed:
Fix bug for node shifting when rtl.

JPJPJPOPOP committed:
Able to delete deprels, move dperls, add root.

JPJPJPOPOP committed:
Editing deprel labels now actually changes labels.

JPJPJPOPOP committed:
Small changes to status.

JPJPJPOPOP committed:
Display
for pos nodes/labels.

JPJPJPOPOP committed:
Fix spacing issues with long pos nodes.

JPJPJPOPOP committed:
Added editing pos labels.

JPJPJPOPOP committed:
Fixed sizing issues when editing pos labels.

JPJPJPOPOP committed:
Added merging of nodes.

JPJPJPOPOP committed:
Add creating supertokens (not displayed on graph yet though).

keggsmurph21 committed:
toggleIsEmpty: Don't automatically set the HEAD to be 'root'

This was a bug I unintentionally introduced when I was adding the
functionality for toggling whether a given token is 'empty'.

keggsmurph21 committed:
toggleIsEmpty: Activate with the "E" key instead of the "N" key

keggsmurph21 committed:
insertEmptyTokenAfter: Add "N" keybinding to insert empty token

keggsmurph21 committed:
package.json: Bump notatrix to v2.3.1

This will allow us to correctly parse MWTs containing empty nodes.

Related to https://github.com/keggsmurph21/notatrix/issues/17.

keggsmurph21 committed:
gui: Refocus #graph after deleting deprel

Since we might be using BACKSPACE to delete the deprel, we want to
make sure that we don't accidentally trigger a "back" page nav event
for the client (a possible default action for Firefox users).

keggsmurph21 committed:
clang-format: Start using an auto-formatter

JPJPJPOPOP committed:
Null nodes now work properly in editor.

JPJPJPOPOP committed:
Make enhanced dependencies display below tokens.

JPJPJPOPOP committed:
Added display of supertokens, and changed height algorithm to account for enhanced deprels being below.

JPJPJPOPOP committed:
Add supertoken label.

JPJPJPOPOP committed:
Added splitting of supertokens.

JPJPJPOPOP committed:
Fix supertoken in rtl and in cg3.

JPJPJPOPOP committed:
Fix issues in Chrome.

JPJPJPOPOP committed:
Added editing forms.

JPJPJPOPOP committed:
Added zooming by +/-/=

JPJPJPOPOP committed:
Fix tabbing through labels and bug with deprel label editting.

JPJPJPOPOP committed:
Fix empty nodes setting and inserting.

JPJPJPOPOP committed:
Remove unnecessary files.

JPJPJPOPOP committed:
Add mouse in collab.

JPJPJPOPOP committed:
Small fix to remove errors when building.

jonorthwash committed:
Merge pull request #411 from JPJPJPOPOP/master

Small fix to remove errors when building.

JPJPJPOPOP committed:
Add back set locks.

JPJPJPOPOP committed:
Fix splitting tokens.

JPJPJPOPOP committed:
Add/fix comments to code.

JPJPJPOPOP committed:
Fix bug in shifting of tokens.

JPJPJPOPOP committed:
Added root deprel.

JPJPJPOPOP committed:
Add export to svg, but export to png does not work.

JPJPJPOPOP committed:
Added export to png. Does not work on Firefox though.

JPJPJPOPOP committed:
Fix export to png on Firefox.

JPJPJPOPOP committed:
Remove more unnecessary files and fix bug with root node deprel.

jonorthwash committed:
Merge pull request #409 from JPJPJPOPOP/port

Porting graph editor to svg / d3js

JPJPJPOPOP committed:
Style changes and change to token numbers.

jonorthwash committed:
Merge pull request #412 from JPJPJPOPOP/stylechanges

Style changes and change to token numbers.

jonorthwash committed:
screenshot with JPJPJPOPOP's latest updates

jonorthwash committed:
screenshot with JPJPJPOPOP's latest updates

jonorthwash committed:
some more updates to README

JPJPJPOPOP committed:
Style changes to token number.

jonorthwash committed:
Merge pull request #413 from JPJPJPOPOP/stylechanges

Style changes to token number.

JPJPJPOPOP committed:
Changes to style and key shortcuts.

jonorthwash committed:
Merge pull request #416 from JPJPJPOPOP/stylechanges

Changes to style and key shortcuts.

keggsmurph21 committed:
package.json: Bump notatrix to v2.3.2

Incorporates changes from https://github.com/keggsmurph21/notatrix/issues/18

JPJPJPOPOP committed:
Add vertical tree view.

JPJPJPOPOP committed:
Fixes to vertical view.

JPJPJPOPOP committed:
Revert code spacing.

JPJPJPOPOP committed:
Add error display for switching to vertical with incomplete graph.

JPJPJPOPOP committed:
Raise root deprel.

JPJPJPOPOP committed:
Make the deprels connecting subtokens dotted.

JPJPJPOPOP committed:
Don't stay in vertical view if warning is given.

jonorthwash committed:
Merge pull request #417 from JPJPJPOPOP/master

Added vertical tree view

keggsmurph21 committed:
notatrix: Cannibalize keggsmurph21/notatrix@9ea6840

keggsmurph21 committed:
notatrix: Upgrade to babel 7+

keggsmurph21 committed:
Switch to using internal "notatrix" rather than from npm

Closes #415
Closes #426

ftyers committed:
Merge pull request #436 from keggsmurph21/merge-notatrix

Merge notatrix

keggsmurph21 committed:
gulp+babel: Remove explicit reference to deprecated package

According to https://babel.dev/docs/en/babel-plugin-transform-property-mutators,
this package is already included in `@babel/preset-env`, so we don't need to
(and shouldn't) install or call it independently.

keggsmurph21 committed:
package.json: Update minor versions of packages

keggsmurph21 committed:
package.json: Bump jsdoc2markdown => 7.x

Apparently, 4.x was broken on newer versions of `node`.  See this[1]
issue for details.

Anyways, as of this commit we should build :^)

[1] https://github.com/hegemonic/requizzle/pull/4

keggsmurph21 committed:
Merge pull request #446 from keggsmurph21/fix/broken-notatrix-build

Fix broken notatrix build

jonorthwash committed:
Merge branch 'master' of github.com:jonorthwash/ud-annotatrix

ftyers committed:
update readme

ftyers committed:
update help text

jonorthwash committed:
now builds without gulp error

Commit:
now builds without gulp error

Commit:
auto punctuation

jonorthwash committed:
auto punctuation

ftyers committed:
fix indexing / help / and hide labels

ftyers committed:
fix indexing / help / and hide labels

keggsmurph21 committed:
gulpfile.js: Fix build-watch command to track notatrix/

ftyers committed:
deploy?

ftyers committed:
deploy?

jonorthwash committed:
redo commit e103f1a16144f5603cb19046ea1f2a65040d4214

jonorthwash committed:
Merge pull request #457 from jonorthwash/bugfix/make-build-watch-track-notatrix

[Take 2] gulpfile.js: Fix `build-watch` command to track `notatrix/`

ftyers committed:
vendor d3

ftyers committed:
Merge branch 'master' of github.com:/jonorthwash/ud-annotatrix

Lguyogiro committed:
fix instructions for creating super token

ftyers committed:
Merge pull request #467 from Lguyogiro/update-supertoken-docs

fix instructions for creating super token

ftyers committed:
Update help.ejs

keggsmurph21 committed:
notatrix: Move token-index extraction to a helper on `BaseToken`

keggsmurph21 committed:
client: Re-use `BaseToken::getTokenIndex()`

keggsmurph21 committed:
notatrix: Extract token-sorting to a helper

keggsmurph21 committed:
notatrix: Check token indices before merging/combining

Closes https://github.com/jonorthwash/ud-annotatrix/issues/475#issuecomment-1073364821

keggsmurph21 committed:
notatrix: Remove a pointless `console.log()`

keggsmurph21 committed:
Bump package-lock and rebuild

keggsmurph21 committed:
notatrix: Prevent combining *into* subtokens

https://github.com/jonorthwash/ud-annotatrix/issues/475#issuecomment-1073363797

jonorthwash committed:
Update README.md

jonorthwash committed:
Update AUTHORS

keggsmurph21 committed:
notatrix: Add support for TypeScript!

At the moment this doesn't do much -- it's just adding support for
running a no-op JavaScript => JavaScript transformation.

Now, if you run

    $ cd notatrix/
    $ npx tsc

you'll see some new files like

    $ tree -F -L 2 build/js/
    build/js/
    ├── data/
    │   ├── apertium-stream.js
    │   ├── brackets.js
    │   ├── cg3.js
    │   ├── conllu.js
    │   ├── index.js
    │   ├── notatrix-serial.js
    │   ├── params.js
    │   ├── plain-text.js
    │   └── sd.js
    └── src/
        ├── converter.js
        ├── detector.js
        ├── formats/
        ├── generator.js
        ├── index.js
        ├── nx/
        ├── parser.js
        ├── splitter.js
        └── utils/

This never actually gets run automatically though!  That will come
later :^)

P.S. - I'm just following this guide:

    https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html

keggsmurph21 committed:
notatrix: Enable `noFallthroughCasesInSwitch` compiler option

https://www.typescriptlang.org/tsconfig#noFallthroughCasesInSwitch

keggsmurph21 committed:
notatrix: Enable `noImplicitReturns` compiler option

https://www.typescriptlang.org/tsconfig#noImplicitReturns

keggsmurph21 committed:
notatrix: Add plugin to automatically run `tsc` as part `gulp`

Now, when we do a "normal"

    npx gulp

we'll automatically call the compiler :tada:

keggsmurph21 committed:
notatrix: Convert first file to TypeScript!

This file just got renamed `constants.{j => ts}` with no changes.

We also had a to add a dependency so that `tsc` would know what
`exports` (a CommonJS[1] thing) means.

[1] https://nodejs.org/api/modules.html#moduleexports

keggsmurph21 committed:
notatrix: Replace decimal constant w hex equivalent

I guess I didn't realize this syntax was legal at the time..?

keggsmurph21 committed:
notatrix: Convert `src/utils/regex.{j => t}s`

keggsmurph21 committed:
notatrix: Convert `src/utils/errors.{j => t}s`

This change actually required making some new type annotations!

keggsmurph21 committed:
notatrix: Convert `src/utils/funcs.{j => t}s`

keggsmurph21 committed:
notatrix: Convert `src/utils/index.{j => t}s`

See [1] for an explanation of why we needed to switch `require` to
`import`.

[1] https://stackoverflow.com/a/35759334

keggsmurph21 committed:
notatrix: Set `prototype` explicitly for `Error` subclasses

Basically, since we're compiling our JavaScript to ES5, we need to
handle this weird edge-case where transitive `extends Error`s don't
work as expected.  This is super arcane and confusing... see [1].

Also, I'm still not sure exactly why, but after a couple hours of
debugging I think this change also requires us to make a tiny update
to the tests??

[1] https://github.com/microsoft/TypeScript/wiki/FAQ#why-doesnt-extending-built-ins-like-error-array-and-map-work

keggsmurph21 committed:
notatrix: Switch the `main` file

Since we'll now we writing TypeScript, we to make sure that consuming
JavaScript libraries (`client/` and `server/`) are looking at `notatrix`'s
generated JavaScript (they don't know how to understand TypeScript).

Eventually, it would probably be better to just make `client/` and
`server/` themselves TypeScript, and then we wouldn't need to rely
on the code being already-compiled when we go to bundle it....

keggsmurph21 committed:
notatrix: Run unit tests via `ts-node`

This library basically provides a `node`-like REPL for our TypeScript
files.  In effect, it lets us "run" the TypeScript files directly,
without needing an explicit compilation step:

    $ cd notatrix/
    $ npx ts-node
    > import * as nx from "./src";

This commit also passes a new flag to our test runner so that it can
take advantage of the same API.

keggsmurph21 committed:
server: Rebuild bundle

keggsmurph21 committed:
notatrix: Switch `npm run build*` to invoke `tsc`, not `gulp`

Since this project is just included in the `client/` and `server/`
packages, there's no reason anyone should need to independently bundle
it.

Also, we forgot to add "scripts" for the new `tsc` stuff, so that's
a convenient thing to replace it with :^)

keggsmurph21 committed:
READMEs: Simplify `notatrix` build/install instructions

keggsmurph21 committed:
notatrix: Remove unused `gulp`-related dependencies

We don't need any of this stuff anymore, since we don't need to produce
an independent bundle of this stuff \o/

keggsmurph21 committed:
package-lock: Bump `lockfileVersion` and rebuild

As of `npm` version 7.0.0 (released 2020-10-13), these files are now
stored with a new schema[1].

I produced this by just running

    $ rm package-lock.json
    $ npm install

[1] https://github.blog/2021-02-02-npm-7-is-now-generally-available/#changes-to-the-lockfile

keggsmurph21 committed:
server: Rebuild bundle

keggsmurph21 committed:
notatrix: Add explicit type annotations to utils

keggsmurph21 committed:
notatrix: Add dev dependency on `@types/underscore` + `@types/uuid`

keggsmurph21 committed:
notatrix: Remove `utils.funcs.dedup()` in favor of `Object` methods

keggsmurph21 committed:
notatrix: Add type annotations for all "fields"

keggsmurph21 committed:
notatrix: Add type annotations for all "splitters"

keggsmurph21 committed:
notatrix: Add type annotations for all "get-loss" calculations

keggsmurph21 committed:
notatrix: Add type annotations for all "generators"

keggsmurph21 committed:
notatrix: Add type annotations for all "detectors"

keggsmurph21 committed:
notatrix: Add type annotations for all "parsers"

keggsmurph21 committed:
notatrix: Glue it all together!

keggsmurph21 committed:
notatrix: Don't `allowJs` anymore :^)

keggsmurph21 committed:
server: Rebuild bundle

keggsmurph21 committed:
notatrix: Add type annotations to all of `nx`

keggsmurph21 committed:
notatrix: Enable `noImplicitAny` compiler option

I've had this toggled on while working on the project so far.  In
my opinion, the stricter we can be, the better :+1:.

keggsmurph21 committed:
package.json: Add `typescript` as dev-dependency to top-level config

We're already using this in `notatrix/package.json`, which is considered
by `npm` to be a distinct project.  Since we also want to add TypeScript
support to `client/` and `server/`, we need this here, too.

keggsmurph21 committed:
client: Remove unused "test" code

This code hasn't been functional since at least 67eb0ff (Jan 2019),
and probably even 0262c13 (Aug 2018).  There are probably _some_ test
cases here that would be worth salvaging, but the vast majority of it
relies on deleted APIs...

If it turns out that we do want to come back and actually run some
tests, I think it'll be easier to start with a clean slate, using the
stuff removed in this patch as inspiration.

keggsmurph21 committed:
client: Add plugin to automatically run `tsc` as part of `gulp`

See da6b331de94e26d28bdb57c14b5d9f2492b4c32b.

keggsmurph21 committed:
client: Convert to TypeScript!

keggsmurph21 committed:
client: Remove autogenerated docs

IMO, autogenerated documentation just doesn't have a lot of value for
our use case -- an application with a single usage.  It's always going
to be more reliable to just check the code.  And, now that we're using
TypeScript, IDEs should be able to help in 99% of cases.

keggsmurph21 committed:
client: Enable `noImplicitAny` compiler option

keggsmurph21 committed:
client: Don't allowJs anymore :^)

keggsmurph21 committed:
server: Rebuild bundle

keggsmurph21 committed:
server: Remove vendored `d3` code

We're now bundling via `browserify` (and retrieving it with `npm`),
so we don't need to hang on to these static copies.

keggsmurph21 committed:
server: Add a basic `tsconfig.json`

This config basically tells `tsc` to *just* type-check our annotations,
but not to actually generate any code (since we only need to run this
locally).

keggsmurph21 committed:
server: Remove unused "old" modals

keggsmurph21 committed:
server: Add type hints!

keggsmurph21 committed:
package.json: Use `ts-node` to `npm run server`

This way, we don't need to run the compiler independently -- we can
just execute the TypeScript files as if they were JavaScript!

keggsmurph21 committed:
package.json: Remove unused `main` key

The file referenced doesn't even exist anymore :^)

IMO, we don't need this key anyway, since we don't expect anyone to
use us as a library.. that wouldn't really make sense.

keggsmurph21 committed:
server: Remove some unused SQLite3 "backend" for corpora

We're always just storing our corpora as JSON, so we don't need any
of this sqlite logic.

In the future, I could imagine it might be interesting to add this
back in, but I think it's better not have to maintain this dead code
in the meantime...

keggsmurph21 committed:
server: Enable `noImplicitAny` compiler option

keggsmurph21 committed:
server: Remove lots of commented-out code

keggsmurph21 committed:
notatrix: Export directly as a TypeScript project

keggsmurph21 committed:
notatrix: Remove unused `jsdoc2md` documentation generator

keggsmurph21 committed:
notatrix: Update READMEs about new `build`-less process

Since `client/` and `server/` are now consuming this project directly
as TypeScript, we don't have a separate build step anymore.

keggsmurph21 committed:
notatrix: Move "data" into `test/` directory

We don't want to make all this random data available in the browser
at runtime...

keggsmurph21 committed:
notatrix: Remove some unused tests

keggsmurph21 committed:
Combine `client/`, `notatrix/` and `server/` projects into `src/`

Functionally, this is equivalent to what we had before, but it makes
it a little easier to share the code into the final bundle (otherwise
the fact that `notatrix` was its "own project" led to annoying errors
with `babel`/`browserify` that didn't seem worth solving...).

keggsmurph21 committed:
server: Rebuild bundle

keggsmurph21 committed:
gitignore: Ignore `.DS_Store`

keggsmurph21 committed:
scripts: Remove the `fail-nicely.*` scripts

keggsmurph21 committed:
test: Fix test imports

Now, this works correctly:

    $ npm run test

keggsmurph21 committed:
resources: Add directory to hold random static assets

keggsmurph21 committed:
server: Skip over malformed corpora

Rather than crashing the whole application, we can just avoid any
files in `corpora_path` that we don't know how to parse.  Obviously,
this isn't great if the file was corrupted somehow (but does contain
data we'd like to keep)...

Tested this by running

    $ cat > corpora/issue506.json
    {"hello":

    $ npm run server  # before
    SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at /home/kevinmurphy/src/jonorthwash/ud-annotatrix/src/server/list-treebanks.ts:51:55
    at Array.map (<anonymous>)
    at /home/kevinmurphy/src/jonorthwash/ud-annotatrix/src/server/list-treebanks.ts:48:34
    at FSReqCallback.oncomplete (fs.js:192:5)

    $ npm run server  # after
    Failed to parse corpora/issue506.json: SyntaxError: Unexpected end of JSON input

Closes https://github.com/jonorthwash/ud-annotatrix/issues/506.

furkan7258 committed:
fix typo: editting -> editing

jonorthwash committed:
disable notatrix and server, add help modal, rebuild?

jonorthwash committed:
how it was

jonorthwash committed:
and putting back again per @ftyers

jonorthwash committed:
help modal

jonorthwash committed:
apparently underscore is needed and not called?
